fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -41,23 +42,22 @@
 import com.ibm.fhir.persistence.context.FHIRPersistenceContextFactory;
 import com.ibm.fhir.persistence.helper.FHIRPersistenceHelper;
 import com.ibm.fhir.persistence.helper.FHIRTransactionHelper;
-import com.ibm.fhir.search.compartment.CompartmentUtil;
 import com.ibm.fhir.search.context.FHIRSearchContext;
 import com.ibm.fhir.search.util.SearchUtil;
 
 /**
  * Bulk patient export Chunk implementation - the Reader.
  *
  */
+@Dependent
 public class ChunkReader extends AbstractItemReader {
     private final static Logger logger = Logger.getLogger(ChunkReader.class.getName());
     protected int pageNum = 1;
-    protected int indexOfCurrentResourceType = 0;
     // Control the number of records to read in each "item".
     protected int pageSize = Constants.DEFAULT_SEARCH_PAGE_SIZE;
 
     protected FHIRPersistence fhirPersistence;
-    protected List<String> resourceTypes;
+    Class<? extends Resource> resourceType;
```

Comment:
protected? 
this change from List is because each resourceType is split into separate executions?

Reply:
yes, not each partition/chuck codes only need to process its own resouretype - one and only one resource type.